tensorflow/compiler/xla/service/llvm_ir/ir_builder_mixin.h

```diff
@@ -71,7 +71,7 @@ class IrBuilderMixin {
 
   template <class... Args>
   llvm::Value* BitCast(Args&&... args) {
-    return mixin_builder()->CreatePointerBitCastOrAddrSpaceCast(std::forward<Args>(args)...);
+    return mixin_builder()->CreaterBitCast(std::forward<Args>(args)...);
```

Comment:
@sumesh13 could you provide rationale for this commit? Are there specific test cases fixed by this?

Upstream code is actually incorrect for AMD GPU.

On AMD GPU, stack variables are by default allocated on `addrspace(5)`, not on default addrspace 0. Therefore I deliberately changed quite a few places in this commit:

https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/3c44312177bbbab3642f145afd96cb7e5e75831a

to cope with the necessity that we sometimes need to do bitcast, and sometimes addrspacecast.

Reply:
The upstream code is likely what needs to check if its a pointer and if it correspondingly call the appropriate builder. In this case, the values are non pointes